From ab84398ed0cbe2a9deb4c178ebef8296ab2db7e3 Mon Sep 17 00:00:00 2001
From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Thu, 12 May 2022 23:25:33 -0300
Subject: [PATCH 313/323] rt: remove extra parameter from __trace_stack()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the build error below while keeping the current PREEMPT_RT code:

kernel/trace/trace_events_trigger.c: In function ‘stacktrace_trigger’:
kernel/trace/trace_events_trigger.c:1227:3: error: too many arguments to function ‘__trace_stack’
   __trace_stack(file->tr, flags, STACK_SKIP, preempt_count());
   ^~~~~~~~~~~~~
In file included from kernel/trace/trace_events_trigger.c:15:
kernel/trace/trace.h:826:6: note: declared here
 void __trace_stack(struct trace_array *tr, unsigned int trace_ctx, int skip);
      ^~~~~~~~~~~~~

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 kernel/trace/trace_events_trigger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/trace/trace_events_trigger.c b/kernel/trace/trace_events_trigger.c
index e4340958da2df..7f25cc48bc503 100644
--- a/kernel/trace/trace_events_trigger.c
+++ b/kernel/trace/trace_events_trigger.c
@@ -1226,7 +1226,7 @@ stacktrace_trigger(struct event_trigger_data *data, void *rec,
 
 	if (file) {
 		local_save_flags(flags);
-		__trace_stack(file->tr, flags, STACK_SKIP, preempt_count());
+		__trace_stack(file->tr, STACK_SKIP, preempt_count());
 	} else
 		trace_dump_stack(STACK_SKIP);
 }
-- 
2.44.0

